libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerRecognizer.cs

```diff
@@ -212,6 +212,8 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dialog
                 recognizerResult.Intents.Add("None", new IntentScore() { Score = 1.0f });
             }
 
+            this.TelemetryClient.TrackEvent("QnAMakerRecognizerResult", this.FillRecognizerResultTelemetryProperties(recognizerResult, telemetryProperties), telemetryMetrics);
```

Comment:
> this.TelemetryClient [](start = 11, length = 21)

Why not dialogContext.TurnState.Get<TelemetryClient>()?

Reply:
As per response above.